git.go

```diff
@@ -22,33 +22,50 @@ func MustRunGitCmd(repoPath string, args ...string) {
 	}
 }
 
-// MustGitCommit stages changes in the dstask repository and commits them. If
-// any error is encountered, the program exits.
+// MustGitCommit is like GitCommit, except if any error is
+// encountered, the program exits.
 func MustGitCommit(repoPath, format string, a ...interface{}) {
 	msg := fmt.Sprintf(format, a...)
 
-	// git add all changed/created files
-	// could optimise this to be given an explicit list of
-	// added/modified/deleted files -- only if slow.
 	fmt.Printf("\n%s\n", msg)
 	fmt.Printf("\033[38;5;245m")
 
+	if err := GitCommit(repoPath, format, a...); err != nil {
+		ExitFail("error: %s", err)
+	}
+
+	fmt.Printf("\033[0m")
+}
+
+// GitCommit stages changes in the dstask repository and commits them.
+func GitCommit(repoPath, format string, a ...interface{}) error {
+	msg := fmt.Sprintf(format, a...)
+
 	// needed before add cmd, see diff-index command
-	bins, _ := ioutil.ReadDir(path.Join(repoPath, ".git/objects"))
+	bins, err := ioutil.ReadDir(path.Join(repoPath, ".git/objects"))
+	if err != nil {
+		return fmt.Errorf("failed to run git commit: %s", err)
+	}
 	brandNew := len(bins) <= 2
 
+	// git add all changed/created files
+	// could optimise this to be given an explicit list of
+	// added/modified/deleted files -- only if slow.
 	// tell git to stage (all) changes
-	MustRunGitCmd(repoPath, "add", ".")
+	if err = RunGitCmd(repoPath, "add", "."); err != nil {
+		return fmt.Errorf("failed to add changes to repo: %s", err)
+	}
```

Comment:
interesting to treat this as an error.
I think there's at least a few callers of MustGitCommit() for which "no change" would be normal, and shouldn't trigger an ExitFail. the taskwarrior and github importers for one, but i think it's also true for many of the commands in commands.go ?

Reply:
I agree having it as an error probably isn't correct, but currently we have no facility for status returned if that makes sense. Just outputting to `stdout` isn't something a lib should do so.
I guess we could just return silently?

Reply:
returning a `nil` error seems appropriate and in line with previous behavior. That's what I would recommend.
regarding disallowing side effects like printing to stdout, it's noble, but not practical. I don't think we should be too purist about it.

Reply:
Yeah, this is fine from a pragmatic standpoint, and nothing to do with being a purist. Remember we are trying to make it more like a library so it can be used in applications outside the CLI

Reply:
Just checking @Dieterbe your referring to the no changes detect right?

Reply:
yeah. i think your last commit is good.
i know the goal is to empower GUI apps. Conceivably we could return a rich response with additional metadata, but before we do that, we'd have to study the call sites to see if this information is even relevant to the end user. i suspect it often isn't.